src/common/src/util/chunk_coalesce.rs

```diff
@@ -256,6 +257,34 @@ impl SlicedDataChunk {
     fn capacity(&self) -> usize {
         self.data_chunk.capacity() - self.offset
     }
+
+    pub fn trunc_data_chunk(
+        data_chunk_builder: &mut DataChunkBuilder,
+        data_chunk: DataChunk,
+    ) -> ArrayResult<Vec<DataChunk>> {
+        let mut sliced_data_chunk = SlicedDataChunk::new_checked(data_chunk)?;
+        let mut res = vec![];
```

Comment:
It defeats the purpose of `SlicedDataChunk` and `yield`. We should yield as soon as one output chunk is available, rather than collecting all of them in a vec.

It is possible to wrap the loop into a utility function like this, but this should also yield output one by one. I am okay with keeping the previous impl and doing the refactor later.